packages/serialization/form/src/formParseNode.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  AdditionalDataHolder,
+  DateOnly,
+  Duration,
+  Parsable,
+  ParsableFactory,
+  ParseNode,
+  TimeOnly,
+  toFirstCharacterUpper,
+} from "@microsoft/kiota-abstractions";
+
+export class FormParseNode implements ParseNode {
+  private readonly _fields: Record<string, string> = {};
+  /**
+   *
+   */
+  constructor(private readonly _rawString: string) {
+    if (!_rawString) {
+      throw new Error("rawString cannot be undefined");
+    }
+    _rawString
+      .split("&")
+      .map((x) => x.split("="))
+      .filter((x) => x.length === 2)
+      .forEach((x) => {
+        const key = this.normalizeKey(x[0]);
+        if (this._fields[key]) {
+          this._fields[key] += "," + x[1];
+        } else {
+          this._fields[key] = x[1];
+        }
+      });
+  }
+  private normalizeKey = (key: string): string => key.trim();
+  public onBeforeAssignFieldValues: ((value: Parsable) => void) | undefined;
+  public onAfterAssignFieldValues: ((value: Parsable) => void) | undefined;
+  public getStringValue = (): string => decodeURIComponent(this._rawString);
+  public getChildNode = (identifier: string): ParseNode | undefined => {
+    if (this._fields[identifier]) {
+      return new FormParseNode(this._fields[identifier]);
+    }
+    return undefined;
+  };
+  public getBooleanValue = (): boolean =>
+    JSON.parse(this.getStringValue()) as boolean;
```

Comment:
Won't this throw an exception when it tries to JSON.parse the _rawString that was passed into the constructor?
Or is the intent that this would only ever be called on a childNode? Although the same issue still arises when the childNode isn't a boolean.

What should be the expected behavior if the parse throws?

Reply:
The fact that there doesn't seem to be a try parse for boolean in JavaScript (true|false|undefined) makes me sad...
I had a look at what I did for the text provider, which was better but partial. So I fixed both implementations. Thanks for calling this out.